packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,507 @@
+    event GuardConfigured(Safe indexed safe, uint256 timelockDelay);
+
+    /// @notice Emitted when a transaction is scheduled for a Safe.
+    /// @param safe The Safe whose transaction is scheduled.
+    /// @param txId The identifier of the scheduled transaction (nonce-independent).
+    /// @param when The timestamp when execution becomes valid.
+    event TransactionScheduled(Safe indexed safe, bytes32 indexed txId, uint256 when);
+
+    /// @notice Emitted when a transaction is cancelled for a Safe.
+    /// @param safe The Safe whose transaction is cancelled.
+    /// @param txId The identifier of the cancelled transaction (nonce-independent).
+    event TransactionCancelled(Safe indexed safe, bytes32 indexed txId);
+
+    /// @notice Emitted when the cancellation threshold is updated
+    event CancellationThresholdUpdated(Safe indexed safe, uint256 oldThreshold, uint256 newThreshold);
+
+    /// @notice Emitted when a transaction is executed for a Safe.
+    /// @param safe The Safe whose transaction is executed.
+    /// @param nonce The nonce of the Safe for the transaction being executed.
+    /// @param txHash The identifier of the executed transaction (nonce-independent).
+    event TransactionExecuted(Safe indexed safe, uint256 indexed nonce, bytes32 txHash);
+
+    ////////////////////////////////////////////////////////////////
+    //                  Internal View Functions                   //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Returns the blocking threshold threshold for a given safe
+    /// @param _safe The Safe address to query
+    /// @return The current blocking threshold
+    function _blockingThreshold(Safe _safe) internal view returns (uint256) {
+        // The blocking threshold is the number of owners who can coordinate to block a transaction
+        // from being executed by refusing to sign.
+        return _safe.getOwners().length - _safe.getThreshold() + 1;
+    }
+
+    /// @notice Returns the cancellation threshold for a given safe
+    /// @param _safe The Safe address to query
+    /// @return The current cancellation threshold
+    function cancellationThresholdForSafe(Safe _safe) public view returns (uint256) {
+        // Return 0 if guard is not enabled
+        if (!_isGuardEnabled(_safe)) {
+            return 0;
+        }
+
+        return _safeState[_safe].cancellationThreshold;
+    }
+
+    /// @notice Returns the maximum cancellation threshold for a given safe
+    /// @param _safe The Safe address to query
+    /// @return The maximum cancellation threshold
+    function maxCancellationThreshold(Safe _safe) public view returns (uint256) {
+        uint256 blockingThreshold = _blockingThreshold(_safe);
+        uint256 quorum = _safe.getThreshold();
+        // Return the minimum of the blocking threshold and the quorum
+        return (blockingThreshold < quorum ? blockingThreshold : quorum) - 1;
```

Comment:
Wrote a long text about why `min(quorum, blocking_threshold)` is a reasonable value for `maxCancellationThreshold`. In short, it opens up to a DoS attack that can be solved with the LivenessModule, but protects us from a quorum-compromised multisig.